block.c

```diff
@@ -538,7 +538,7 @@ ATTR_NONNULL_ALL void block_show_status(int fd, ddhcp_config* config) {
 
     if (timeout > 0) {
       num_reserved_blocks++;
-      dprintf(fd, "%i\t%i\t%s\t%u\t%s\t%lu\n", block->index, block->state, node_id, block->claiming_counts, leases, timeout);
+      dprintf(fd, "%i\t%i\t%s\t%u\t%s\t%jd\n", block->index, block->state, node_id, block->claiming_counts, leases, (intmax_t) timeout);
```

Comment:
I looked a bit more into the portability implications and found that: https://stackoverflow.com/questions/18839043/format-specifiers-for-implementation-defined-types-like-time-t

It seems using difftime to calculate the timeout and then using a format string for double is the best way.